lib/stableSample.js

```diff
@@ -0,0 +1,77 @@
+const {Cu} = require('chrome');
+Cu.importGlobalProperties(['crypto', 'TextEncoder']);
+
+const {Log} = require('./Log.js');
+
+/**
+ * Map from the range [0, 1] to [0, max(sha256)].
+ * @param  {number} frac A float from 0.0 to 1.0.
+ * @return {string} A string in the sha256 hash space. Will be zero padded to 64
+ *   characters.
+ */
+function fractionToKey(frac) {
+  // SHA 256 hashes are 64-digit hexadecimal numbers. The largest possible SHA
+  // 256 hash is 2^256 - 1.
+
+  if (frac < 0 || frac > 1) {
+    throw new Error(`frac must be between 0 and 1 inclusive (got ${frac})`);
+  }
+
+  const mult = Math.pow(2, 256) - 1;
+  const inDecimal = Math.floor(frac * mult);
```

Comment:
I think this is fine, but I want to double check. Both this and the self-repair version are going to have precision errors in the multiplication. The least significant digits will be wrong, and we'll have big jumps. (like 0.011 straight to 0.032 with nothing in between, but a lot smaller numbers).

I think the precision errors all happen at a scale smaller than we care about. What do you think?